flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -315,10 +315,12 @@ class ScalarExprLowering {
                               Fortran::lower::AbstractConverter &converter,
                               Fortran::lower::SymMap &symMap,
                               Fortran::lower::StatementContext &stmtCtx,
-                              bool initializer = false)
+                              bool initializer = false, bool getRawVals = false)
       : location{loc}, converter{converter},
         builder{converter.getFirOpBuilder()}, stmtCtx{stmtCtx}, symMap{symMap},
-        inInitializer{initializer} {}
+        inInitializer{initializer}, genRawVals{getRawVals} {
+    rawType = 0;
```

Comment:
I think you don't need this. `mlir::Type` default constructor will just use a nullptr.